synapse/models/files.py

```diff
@@ -699,6 +702,44 @@ def getModelDefs(self):
                         'doc': 'The file offset to the beginning of the section'}),
                 )),
 
+                ('file:mime:lnk', {}, (
+                    ('file', ('file:bytes', {}), {
+                        'doc': 'The actual file that this shortcut file executed.'}),
+                    ('entry:primary', ('file:path', {}), {
+                        'doc': "The reconstructed primary path of the file this shortcut's FileEntry structure  points to."}),
+                    ('entry:secondary', ('file:path', {}), {
+                        'doc': "The reconstructed secondary path of the file this shortcut's FileEntry structure points to."}),
+                    ('entry:extended', ('file:path', {}), {
+                        'doc': "The target file path from the extended FileEntry structure."}),
+                    ('entry:localized', ('file:path', {}), {
+                        'doc': "The localized target file path from the extended FileEntry structure."}),
+                    ('entry:icon', ('file:path', {}), {
+                        'doc': 'The path to the icon file used when displaying a shortcut file to a user.'}),
+                    ('environment:path', ('file:path', {}), {
+                        'doc': 'The path to the target file when the LNK says to prefer environment variables.'}),
+                    ('environment:icon', ('file:path', {}), {
+                        'doc': 'The path to the icon file displayed when the LNK flags says to prefer environment variables.'}),
+                    ('working', ('file:path', {}), {
+                        'doc': 'The working directory used when activating the link target.'}),
+                    ('relative', ('file:path', {}), {
+                        'doc': 'The path to the target file used when the LNK flags say to use a relative path.'}),
+                    ('arguments', ('it:cmd', {}), {
+                        'doc': 'The command line arguments passed to the target file when the shortcut is activated.'}),
+                    ('desc', ('str', {}), {
+                        'doc': 'The description of the shortcut from the StringData section of the LNK file.'}),
+                    ('link:flags', ('int', {}), {
```

Comment:
Should this just be `flags` since it's the only one in it's prefix and seems like a top level flags entry? ( i could be misinterpreting the use case :D )

Reply:
No you're right. It is a top level flags entry. This was more preemptive namespacing. Because you have the target flags that also comes from the same header, a couple of the other structures have similar flags/typeid structures we may want later down the line, and then there's the mess that is the property store.

I think there's some less confusing names I could go with after going through their mess of docs for this and the property store stuff.